### PR TITLE
Make sure tokenranges are not created when not storing tokens.

### LINKED ISSUE #1149: Let storeTokens also influence tokenrange creation.

### PR DESCRIPTION
**Pull request by matozoid**
Fixes #1149 . @arturbosch can you take a look?

### CODE REVIEW

File: javaparser-testing/src/test/java/com/github/javaparser/ParserConfigurationTest.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import static com.github.javaparser.Providers.provider;
+import static org.junit.Assert.assertFalse;
+
+public class ParserConfigurationTest {
+    @Test
+    public void storeNoTokens() {
+        ParseResult<CompilationUnit> result = new JavaParser(new ParserConfiguration().setStoreTokens(false)).parse(ParseStart.COMPILATION_UNIT, provider("class X{}"));
+        
+        assertFalse(result.getTokens().isPresent());
+        assertFalse(result.getResult().get().getTokenRange().isPresent());
```
Comment (arturbosch) [inline javaparser-testing/src/test/java/com/github/javaparser/ParserConfigurationTest.java]: Maybe add something like `result.getResult.get().getNodesByType(Node.class).stream().allMatch(node -> !node.getTokenRange().isPresent())` (with the premise of extending the test code :))?

**Review by arturbosch — COMMENTED** [review]
The changes look okay to me.

File: javaparser-testing/src/test/java/com/github/javaparser/ParserConfigurationTest.java (hunk 1 above)
Comment (matozoid) [reply javaparser-testing/src/test/java/com/github/javaparser/ParserConfigurationTest.java]: Done!

**Comment by matozoid** [comment]
This one is still not working well...

**Comment by arturbosch** [comment]
What do you mean? Are there still JavaToken objects?

**Comment by matozoid** [comment]
No, the grammar assumes there are always token ranges and runs into endless nullpointers when parsing more complex classes. Working on it, but it will not only drop `TokenRange` but also `Range`.

**Pull request merged**